Django 1.9 compatibility

gavinwahl opened the pull request:
Still waiting on treebeard:

> ImportError raised when trying to load 'treebeard.templatetags.admin_tree': cannot import name EMPTY_CHANGELIST_VALUE

gavinwahl committed:
Django 1.9 compatibility

Commit:
Move thumbnail_libs to page_builder

Django 1.9 loads templatetags on template configuration. Since widgy
itself doesn't depend on sorl thumbnail, and page_builder does, it
belongs there.

See (https://docs.djangoproject.com/en/dev/releases/1.9/#template-tag-modules-are-imported-when-templates-are-configured)

Commit:
Use django.contrib.admin.utils where possible

django.contrib.admin.util is gone in Django 1.9

Commit:
Drop Django 1.7 testing, add Django 1.9 testing

Commit:
Remove custom render_to_string method

This method was only needed by old versions of Django

Commit:
Remove lines of code as instructed by comments.

Commit:
Remove code for old versions of Django

Commit:
Switch django.utils.unittest to unittest

The old django.contrib.unittest has been removed and is not needed in
Django 1.7+

Commit:
Update admin jQuery url for 1.9

Also override admin jQuery in Widgy for backwards compatibility. The
included jQuery should be deleted as soon as Django 1.8 support is
dropped and can most likely be safely updated if necessary for Django 1.10
or Django 1.11.

Commit:
Switch DateTimeField to SplitDateTimeField

DateTimeField no longer supports the SplitDateTime widget

Commit:
Fix ReviewQueueViews test for Django 1.9

Location header is no longer absolute:
https://docs.djangoproject.com/en/1.9/releases/1.9/#http-redirects-no-longer-forced-to-absolute-uris

Commit:
Add easy_thumbnails to contrib test INSTALLED_APPS

Django 1.9 needs apps with models to be installed.

Commit:
Use python importlib for Django >= 1.9

Commit:
Fix ReviewQueue tests for Django 1.9

See
https://github.com/django/django/blob/master/django/db/models/fields/reverse_related.py#L60

Commit:
Fix popup_base split date time rendering

Need to use the AdminSplitDateTime widgets

julianandrews commented:
Now passing all tests with https://github.com/tabo/django-treebeard/pull/25 and https://github.com/stephenmcd/mezzanine/pull/1481 installed.

Mezzanine will likely get merged over Christmas.

There are likely some admin css/js issues still unaddressed.

Commit:
Set Django version for tox to current 1.9

acatton committed:
Fix django-storages compatibility

acatton committed:
Fix the "View live version" link for widgy pages.

This is a port of
https://github.com/stephenmcd/grappelli-safe/commit/6880787b57fa1989e9a8086f811d8d95a4b0e8ca

acatton committed:
The default behavior in filer has been fixed

Commit:
Fix HandleFormMixin to work if form isn't present.

widgy-blog uses this mixin for blog entries which may have no form at
all. For Django LTE 1.8 this worked, but since
https://github.com/django/django/commit/8a1824d465a551b44376f17c01ef142cc036e863
get_form is called by get_context_data and this all breaks.

gavinwahl committed:
WidgyPage needs to be swappable in the migration

Otherwise a useless table gets created

Commit:
Remove --process-dependency-links flag from tox

I added this as part of a hack, and never meant to commit the change!

Commit:
Update broken font-awesome classes

`icon-pencil` -> `fa fa-pencil`
`icon-remove` -> `fa fa-remove`
Also delete unused font-awesome-ie7.css.
